### ISSUE TITLE
Add Ansi coloring printer

### ISSUE THREAD

**Issue by arichiardi** [bhb/expound#44]
Awesome awesome project!

For outputing to terminal, say for instrument called within `clojure.test`, it would be awesome to have ansi color codes.

Just an idea,  keep up the good work and thanks!

**Comment by bhb** [comment]
@arichiardi Great idea! I have a few ideas on how to start with this.

Unfortunately, it's a bit tricky to describe via GitHub issues because AFAICT, we can't colorize the Github-flavored-markdown, but I'll do my best :)

From the sample output in the README:

```
;; -- Spec failed --------------------

;;   {:city ..., :state :CO}
;;                      ^^^

;; should satisfy

;;   string?

;; -- Relevant specs -------

;; :example.place/state:
;;   clojure.core/string?
;; :example/place:
;;   (clojure.spec.alpha/keys
;;    :req-un
;;    [:example.place/city :example.place/state])

;; -------------------------
;; Detected 1 error
```

`:CO` i.e. the "non-conforming value" : color A
`^^^` i.e. the the "highlight" : color B
`string?` i.e. the "predicate" : color A (so it visually matches the non-conforming value).

everything else could remain uncolored for now.

Some more examples of problems:

| Full problem text  | colored  |  uncolored |
|---|---|---|
|  "should be one of: `:bar`,`:foo`" | "`:bar`,`:foo`"  |  "should be one of:" |
|  should have additional elements. The next element is named `:v` and satisfies any? | "The next element is named `:v` and satisfies any?" | "should have additional elements."  |
| "Value has extra input"  | "Value has extra input"  |  n/a |
|  "should contain keys: `:keys-spec/name`,`:age`" |  "`:keys-spec/name`,`:age`" |  "should contain keys:"  |

I suspect that this scheme would help visually connect the non-conforming value with the problem, but I could be wrong - perhaps in practice, it's nicer to have the entire problem (the whole left hand column) colorized.

What do you think? I'd prefer to start with minimal colors and see if it solves the problem. We can always add more colors as we go.

PRs welcome on this. Otherwise, I'll probably tackle after a few more bug fixes and one more big feature I want to work on.

**Comment by bhb** [comment]
> the separator between spec in order to create a visual barrier between different spec problem

Yep, good idea. Probably a third color C for this.

> the text of the second error in the table could be probably colored less 😄

Yes, perhaps just `:v` and `any`.

**Comment by carocad** [comment]
@bhb maybe it would be a good idea to have a color convention similar or equal to `figwheel` colors for compiler error messages.

The idea is to make the colors as predictable as possible, and since figwheel is (afaik) the standard for Clojurescript I think that it would help to ease understanding the error messages.

- [figwheel function](https://github.com/bhauman/lein-figwheel/blob/196b8eff2796fddfd8d68e37c148fbf80fa4fb30/sidecar/src/figwheel_sidecar/components/cljs_autobuild.clj#L44)

- [figwheel feature description](https://github.com/bhauman/lein-figwheel/blob/e47da1658a716f83888e5a5164ee88e59b2d8c1e/sidecar/resources/conf-fig-docs/FigwheelOptions.txt#L136)

Hope it helps

**Comment by bhb** [comment]
@carocad That's a good idea - Figwheel has already put a lot of thought into this, and it would mean that if you're familiar with Figwheel messages, you'l have an easier time reading Expound error messages as well.

**Comment by arichiardi** [comment]
This is great and I will definitely try this out as soon as possible. Thanks a lot!

**Comment by carocad** [comment]
@bhb I just gave it a try and I think it is a wonderful piece of work :)

I will stay with the snapshot version until a full release comes out just so that I can keep the colored output. 

For those that come here and are curious about it here are some screenshots on the output on my Cursive IDE.

![screen shot 2018-04-03 at 11 18 31](https://user-images.githubusercontent.com/10408729/38241109-b420d62a-3731-11e8-9702-f652d9e40d2d.png)
![screen shot 2018-04-03 at 11 18 44](https://user-images.githubusercontent.com/10408729/38241119-ba4ce30e-3731-11e8-8046-41b8560dee8e.png

**Comment by bhb** [comment]
@carocad Thank you for trying it out and for the suggestions above!

**Comment by bhb** [comment]
@ arichiardi @carocad This feature is included in 0.6.0. Thanks for all your help!
